Sources/SwiftInspectorVisitors/Tests/TypeDescriptionSpec.swift

```diff
@@ -0,0 +1,487 @@
+      }
+
+      context("when encoding a tuple type") {
+        beforeEach {
+          sut = .tuple([.simple(name: "Foo"), .optional(.simple(name: "Bar"))])
+          data = "{\"caseDescription\":\"tuple\",\"typeDescriptions\":[{\"caseDescription\":\"simple\",\"text\":\"Foo\"},{\"caseDescription\":\"optional\",\"typeDescription\":{\"caseDescription\":\"simple\",\"text\":\"Bar\"}}]}".data(using: .utf8)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try encoder.encode(sut)) == data
+        }
+      }
+
+      context("when encoding an unknown type") {
+        beforeEach {
+          sut = .unknown(text: "Foo")
+          data = "{\"caseDescription\":\"unknown\",\"text\":\"Foo\"}".data(using: .utf8)
+        }
+
+        it("to decode the encoded type description") {
+          expect(try encoder.encode(sut)) == data
+        }
+      }
+    }
+
+    describe("typeDescription") {
+      context("when called on a TypeSyntax node representing a SimpleTypeIdentifierSyntax") {
+        final class SimpleTypeIdentifierSyntaxVisitor: SyntaxVisitor {
+          var simpleTypeIdentifier: TypeDescription?
+          override func visit(_ node: SimpleTypeIdentifierSyntax) -> SyntaxVisitorContinueKind {
+            simpleTypeIdentifier = TypeSyntax(node).typeDescription
+            return .skipChildren
+          }
+        }
+
+        var visitor: SimpleTypeIdentifierSyntaxVisitor!
+        beforeEach {
+          let content = """
+              var int: Int = 1
+              """
+
+          visitor = SimpleTypeIdentifierSyntaxVisitor()
+          try? VisitorExecutor.walkVisitor(visitor, overContent: content)
+        }
+
+        it("Finds the type") {
+          expect(visitor?.simpleTypeIdentifier?.asSource) == "Int"
+        }
+      }
+
+      context("when called on a TypeSyntax node representing a MemberTypeIdentifierSyntax") {
+        final class MemberTypeIdentifierSyntaxVisitor: SyntaxVisitor {
+          var nestedTypeIdentifier: TypeDescription?
+          override func visit(_ node: MemberTypeIdentifierSyntax) -> SyntaxVisitorContinueKind {
+            nestedTypeIdentifier = TypeSyntax(node).typeDescription
+            return .skipChildren
+          }
+        }
+
+        var visitor: MemberTypeIdentifierSyntaxVisitor!
+        beforeEach {
+          let content = """
+              var int: Swift.Int = 1
+              """
+
+          visitor = MemberTypeIdentifierSyntaxVisitor()
+          try? VisitorExecutor.walkVisitor(visitor, overContent: content)
+        }
+
+        it("Finds the type") {
+          expect(visitor?.nestedTypeIdentifier?.asSource) == "Swift.Int"
+        }
+      }
+
+      context("when called on a TypeSyntax node representing a CompositionTypeSyntax") {
+        final class CompositionTypeSyntaxVisitor: SyntaxVisitor {
+          var composedTypeIdentifier: TypeDescription?
+          // Note: ideally we'd visit a node of type CompositionTypeElementListSyntax
```

Comment:
`CompositionTypeElementListSyntax` is a list of composed types, where a list of composed types looks like `Foo & Bar`.

`InheritedTypeSyntax` tells us what type(s) a type declaration inherits from. So in this case, the inherited type of `protocol FooBar` is `Foo & Bar`. And since `InheritedTypeSyntax` has a `TypeSyntax` property, we can easily call `typeDescription` to determine the underlying type description.

Lists on the other hand don't have a way to access the types in the list without utilizing a `SyntaxVisitor`. Hence the use of `InheritedTypeSyntax` and the corresponding note 🙂